src/dmd/backend/symbol.d

```diff
@@ -0,0 +1,2454 @@
+                    list_dehydrate(&tm.TMnestedfriends,cast(list_free_fp)&tmnf_dehydrate);
+                    param_dehydrate(&tm.TMptpl);
+                    param_dehydrate(&tm.TMptal);
+                    token_dehydrate(&tm.TMbody);
+                    list_dehydrate(&tm.TMmemberfuncs,cast(list_free_fp)&tmf_dehydrate);
+                    list_dehydrate(&tm.TMexplicit,cast(list_free_fp)&tme_dehydrate);
+                    list_dehydrate(&tm.TMnestedexplicit,cast(list_free_fp)&tmne_dehydrate);
+                    ph_dehydrate(&tm.TMnext);
+                    symbol_dehydrate(&tm.TMpartial);
+                    symbol_dehydrate(&tm.TMprimary);
+                }
+                break;
+            }
+
+            case SCnamespace:
+                symbol_tree_dehydrate(&s.Snameroot);
+                list_dehydrate(&s.Susing,cast(list_free_fp)&symbol_dehydrate);
+                break;
+
+            case SCmemalias:
+            case SCfuncalias:
+            case SCadl:
+                list_dehydrate(&s.Spath,cast(list_free_fp)&symbol_dehydrate);
+            case SCalias:
+                ph_dehydrate(&s.Smemalias);
+                break;
+
+            default:
+                if (s.Sflags & (SFLvalue | SFLdtorexp))
+                    el_dehydrate(&s.Svalue);
+                break;
+        }
+        {   dt_t **pdt;
+            dt_t *dt;
+
+            for (pdt = &s.Sdt;
+                 (dt = *pdt) != null && !isdehydrated(dt);
+                 pdt = &dt.DTnext)
+            {
+                ph_dehydrate(pdt);
+                switch (dt.dt)
+                {   case DT_abytes:
+                    case DT_nbytes:
+                        ph_dehydrate(&dt.DTpbytes);
+                        break;
+                    case DT_xoff:
+                        symbol_dehydrate(&dt.DTsym);
+                        break;
+                }
+            }
+        }
+        if (s.Scover)
+            symbol_dehydrate(&s.Scover);
+    }
+}
+}
+
+/***************************
+ * Dehydrate threaded list of symbols.
+ */
+
+static if (DEHYDRATE)
+{
+void symbol_symdefs_dehydrate(Symbol **ps)
+{
+    Symbol *s;
+
+    for (; *ps; ps = &s.Snext)
+    {
+        s = *ps;
+        symbol_debug(s);
+        //printf("symbol_symdefs_dehydrate(%p, '%s')\n",s,s.Sident.ptr);
+        symbol_dehydrate(ps);
+    }
+}
+}
+
+
+/***************************
+ * Hydrate threaded list of symbols.
+ * Input:
+ *      *psx    start of threaded list
+ *      *parent root of symbol table to add symbol into
+ *      flag    !=0 means add onto existing stuff
+ *              0 means hydrate in place
+ */
+
+version (SCPP_HTOD)
+{
+
+void symbol_symdefs_hydrate(Symbol **psx,Symbol **parent,int flag)
+{   Symbol *s;
+
+    //printf("symbol_symdefs_hydrate(flag = %d)\n",flag);
+debug
+{
+    int count = 0;
+
+    if (flag) symbol_tree_check(*parent);
+}
+    for (; *psx; psx = &s.Snext)
+    {
+        //printf("%p ",*psx);
+debug
```

Comment:
Yuck! Needs curly braces.

Reply:
I try to minimize diffs, leaving reformatting for another PR.